Задание 1

In [2]:
import numpy as np
import pandas as pd
import statsmodels.formula.api as smf


In [3]:
df = pd.read_csv('wage2.csv')
df.shape

(935, 17)

In [4]:
 mod = smf.ols(formula = 'np.log(wage)~1+IQ+south' , data = df)
res = mod.fit()

In [5]:
 res.params.round(2)

Intercept    6.01
IQ           0.01
south       -0.12
dtype: float64

Задание 2

In [6]:
# импорт данных
df = pd.read_csv('Labour.csv')
df

capital  labour     output       wage
0     2.606563     184   9.250759  43.080307
1     1.323237      91   3.664310  27.780016
2    22.093692     426  28.781516  44.467748
3    10.737851      72   4.124642  39.734710
4     1.161365      46   2.890150  34.650709
..         ...     ...        ...        ...
564   2.625403      20   1.424376  33.477545
565   1.276386      61   2.109048  26.300732
566   1.953869     117   6.241870  41.153979
567   1.318527      46   7.902237  66.720139
568   2.832391     161   6.804085  36.618435

[569 rows x 4 columns]

In [7]:
mod = smf.ols(formula = 'np.log(output)~1+np.log(capital)+np.log(wage)' , data = df)
res = mod.fit()

In [8]:
res.params

Intercept         -1.942866
np.log(capital)    0.511506
np.log(wage)       0.882337
dtype: float64

In [9]:
# R^2
res.rsquared.round(3)

0.611

In [10]:
 # adj R^2
res.rsquared_adj.round(3)

0.609

ИСПРАВЛЕНО: 
Коэффициент детерминации равен 0.611. Такая доля показателя *выпуска* зависит от капитала компании (в миллионах евро) и от зарплаты на одного сотрудника (в тыс евро).

Задание 3

In [17]:
# импорт данных
df = pd.read_csv('Labour.csv')
df

capital  labour     output       wage
0     2.606563     184   9.250759  43.080307
1     1.323237      91   3.664310  27.780016
2    22.093692     426  28.781516  44.467748
3    10.737851      72   4.124642  39.734710
4     1.161365      46   2.890150  34.650709
..         ...     ...        ...        ...
564   2.625403      20   1.424376  33.477545
565   1.276386      61   2.109048  26.300732
566   1.953869     117   6.241870  41.153979
567   1.318527      46   7.902237  66.720139
568   2.832391     161   6.804085  36.618435

[569 rows x 4 columns]

In [11]:
mod = smf.ols(formula = 'np.log(output)~1+np.log(capital)+np.log(wage)' , data = df)
res = mod.fit()

In [12]:
res.params

Intercept         -1.942866
np.log(capital)    0.511506
np.log(wage)       0.882337
dtype: float64

In [13]:
# TSS
res.centered_tss.round(2)

797.67

In [14]:
# ESS
res.ess.round(2)

487.23

In [15]:
# RSS
res.ssr.round(2)

310.43

Задание 4

In [16]:
# импорт данных
df = pd.read_csv('Labour.csv')
df

capital  labour     output       wage
0     2.606563     184   9.250759  43.080307
1     1.323237      91   3.664310  27.780016
2    22.093692     426  28.781516  44.467748
3    10.737851      72   4.124642  39.734710
4     1.161365      46   2.890150  34.650709
..         ...     ...        ...        ...
564   2.625403      20   1.424376  33.477545
565   1.276386      61   2.109048  26.300732
566   1.953869     117   6.241870  41.153979
567   1.318527      46   7.902237  66.720139
568   2.832391     161   6.804085  36.618435

[569 rows x 4 columns]

In [18]:
mod = smf.ols(formula = 'np.log(output)~1+np.log(capital)+np.log(labour)+np.log(wage)' , data = df)
res = mod.fit()

In [19]:
res.params

Intercept         -5.007316
np.log(capital)    0.149343
np.log(labour)     0.720410
np.log(wage)       0.921390
dtype: float64

In [20]:
 # задаем индексы наблюдений
ind = [46]

In [21]:
# Наблюдаемое значение
np.log(df['output']).iloc[ind].round(3)

46    3.247
Name: output, dtype: float64

In [22]:
# Предсказанное значение
res.fittedvalues[ind].round(3)

46    3.399
dtype: float64

In [23]:
# Остаток
res.resid[ind].round(3)

46   -0.152
dtype: float64

Задание 5

In [24]:
# импорт данных
df = pd.read_csv('Labour.csv')
df

capital  labour     output       wage
0     2.606563     184   9.250759  43.080307
1     1.323237      91   3.664310  27.780016
2    22.093692     426  28.781516  44.467748
3    10.737851      72   4.124642  39.734710
4     1.161365      46   2.890150  34.650709
..         ...     ...        ...        ...
564   2.625403      20   1.424376  33.477545
565   1.276386      61   2.109048  26.300732
566   1.953869     117   6.241870  41.153979
567   1.318527      46   7.902237  66.720139
568   2.832391     161   6.804085  36.618435

[569 rows x 4 columns]

In [25]:
mod = smf.ols(formula = 'np.log(output)~1+np.log(capital)+labour+wage' , data = df)
res = mod.fit()

In [26]:
res.params

Intercept          0.577725
np.log(capital)    0.442294
labour             0.000450
wage               0.016260
dtype: float64

output - зависимая переменная

capital, labour, wage - регрессоры 

output - выпуск (в миллионах евро)

capital - капитал (в миллионах евро)

labour - число сотрудников 

wage - зарплата на одного сотрудника (в тыс евро)

При увеличении значения регрессора capital на единицу при прочих равных, переменная output изменится на 0.44

При увеличении значения регрессора labour на единицу при прочих равных, переменная output изменится на 0.05 (коэффициент, умноженный на 100)

При увеличении значения регрессора wage на единицу при прочих равных, переменная output изменится на 1.63 (коэффициент, умноженный на 100)

In [10]:
import turtle as t
t.shape('turtle')
t.bgcolor('black')
t.pencolor('red')
t.speed(99999)
n=1000
m=6
for k in range(n):
    t.fd(k*((k+n)/n))
    t.rt((360/m)+0.5)
t.exitonclick()
#не смогла вспомнить правильные коэффициенты, но должна быть красивая спиралька:(